pkg/common/adaptor/request_test.go

```diff
@@ -77,16 +90,16 @@ func makeACall(t *testing.T, method, url string, header http.Header, body string
 	if err != nil {
 		t.Fatalf("Read body error: %s", err)
 	}
-	if string(b) != body {
-		t.Fatalf("Body not equal: want: %s, got: %s", body, string(b))
-	}
+	assert.DeepEqual(t, body, string(b))
+	assert.DeepEqual(t, expectStatusCode, resp.StatusCode)
 
-	if resp.StatusCode != expectStatusCode {
-		t.Fatalf("Status code not equal: want: %d, got: %d", expectStatusCode, resp.StatusCode)
-	}
+	// Parse out the cookie to verify it is correct
+	cookie := protocol.Cookie{}
+	_ = cookie.Parse(header[consts.HeaderSetCookie][0])
+	assert.DeepEqual(t, expectCookieValue, cookie.Value())
 }
 
-func handlerAndCheck(t *testing.T, writer http.ResponseWriter, request *http.Request, wantHeader http.Header, wantBody string, statusCode int) {
+func handlerAndCheck(t *testing.T, writer http.ResponseWriter, request *http.Request, wantHeader http.Header, wantBody string, statusCode ...int) {
```

Comment:
What is the purpose of this change: `statusCode` int -> ...int?

Reply:
> What is the purpose of this change: `statusCode` int -> ...int?

Changed to make this parameter optional. When testing the code before this change, a meaningless number 0 will be passed in and judged in the subsequent logic. I think this change can make the code more elegant

Before
```go
handlerAndCheck(t, resp, req, testHeader, testBody, 0)
```
Now
```go
handlerAndCheck(t, resp, req, testHeader, testBody)
```

Before
```go
if statusCode != 0 {
    writer.WriteHeader(statusCode)
}
```

Now
```go
if statusCode != nil {
    writer.WriteHeader(statusCode[0])
}
```

Reply:
From a function signature perspective, it's a little bit strange, maybe add some comment to introduce?

Reply:
> From a function signature perspective, it's a little bit strange, maybe add some comment to introduce?

Done.